prof.c

```diff
@@ -853,30 +935,22 @@ void *utracy_server_thread_start(void *user) {
 		(void) utracy_write(&srcloc.color, sizeof(srcloc.color));
 	}
 
-	int quitting = 0;
+	bool quitting = false;
 	while(!quitting) {
 		struct event evt;
 		while(0 == event_queue_pop(&evt)) {
 			(void) utracy_write(&evt, sizeof(evt));
 		}
-
-#if defined(UTRACY_WINDOWS)
-		switch (WaitForSingleObject(utracy.quit, 1)) {
-		default:
-			LOG_DEBUG_ERROR;
-		case WAIT_OBJECT_0:
-			quitting = 1;
-			break;
-		case WAIT_TIMEOUT:
-			break;
-		}
-#elif defined(UTRACY_LINUX)
-		if (utracy.quit) {
-			break;
+		switch (wait_for_event_pipe(utracy.quit, 1)) {
+			case 0:
+				quitting = true;
+				break;
+			case 1:
+				break;
+			default:
+				LOG_DEBUG_ERROR;
+				break;
 		}
-
-		usleep(1000);
```

Comment:
I see you removed the sleep here- is there anything that sleeps still? Was this changed to no longer require sleeping?

Reply:
From what I can tell, it was only sleeping due to the shoddy linux impl of this (using a `volatile int` rather than proper synchronization) - the `event_pipe` abstraction now means it'll use proper synchronization methods on both Windows and Linux, rather than just on Windows as before.